src/Matrix.jl

```diff
@@ -6503,6 +6503,28 @@ function zero_matrix(R::NCRing, r::Int, c::Int)
    return z
 end
 
+################################################################################
+#
+#   Ones matrix
+#
+################################################################################
+
+@doc raw"""
+    ones_matrix(R::Ring, r::Int, c::Int)
+
+Return the $r \times c$ ones matrix over $R$.
+"""
+function ones_matrix(R::NCRing, r::Int, c::Int)
+   arr = Matrix{elem_type(R)}(undef, r, c)
```

Comment:
```suggestion
   arr = zero_matrix(R, r, c)
```
and then return `arr` at the end. Not sure if this is what @fingolfin had in mind, but it should work.

Reply:
This should works, but wouldn't an implementation of `ones_matrix` for `QQMatrix` and `ZZMatrix` in Nemo be preferable?

Reply:
This change here should be applied in any case. Adding special implementations in Nemo can only be done later

Reply:
I concur with the solution proposed by @thofma here, it seems to be the best we can do *right now*. And @lgoettgens of course also got it right: we first need a generic implementation here, then we can add specialized ones elsewhere.

All in all there are other function in here which ought to be more generic. I think it is time to revive PR #558 . I am looking into that right now.

But that shouldn't delay this PR. Just maybe also mention the new function in a `@docs` block in `docs/src/matrix.md` near `identity_matrix` or so